hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataNode.java

```diff
@@ -1935,6 +1937,10 @@ public void shutdown() {
       }
     }
 
+    if (storageLocationChecker != null) {
+      storageLocationChecker.shutdownAndWait(10, TimeUnit.SECONDS);
```

Comment:
Something for you to consider, most of the other threads seems to shutdown much quicker, do we need to wait 10 seconds ?